plugins/inputs/execd/execd_unix.go

```diff
@@ -0,0 +1,33 @@
+// +build !windows
+
+package execd
+
+import (
+	"fmt"
+	"io"
+	"syscall"
+
+	"github.com/influxdata/telegraf"
+)
+
+func (e *Execd) Gather(acc telegraf.Accumulator) error {
+	if e.cmd == nil || e.cmd.Process == nil {
+		return nil
```

Comment:
Let's move the restart handling to here, like so:
```suggestion
    err := e.cmdRun()
    if err != nil {
      return err
    }
    return nil
```

Same in the Windows version

Reply:
I would need to run `go e.cmdRun()`. How does this work with the error handling?

Reply:
Restart handling has been moved into a standalone goroutine to avoid a race condition in 0e67fe56.